algorithms/src/snark/varuna/ahp/prover/round_functions/second.rs

```diff
@@ -49,44 +42,40 @@ impl<F: PrimeField, MM: SNARKMode> AHPForR1CS<F, MM> {
     pub fn second_round_polynomial_info() -> BTreeMap<PolynomialLabel, PolynomialInfo> {
         [PolynomialInfo::new("h_0".into(), None, None)].into_iter().map(|info| (info.label().into(), info)).collect()
     }
+}
 
-    /// Output the second round message and the next state.
-    pub fn prover_second_round<'a, R: RngCore>(
-        verifier_message: &verifier::FirstMessage<F>,
-        mut state: prover::State<'a, F, MM>,
-        _r: &mut R,
-    ) -> Result<(prover::SecondOracles<F>, prover::State<'a, F, MM>)> {
+impl<'a, F: PrimeField, MM: SNARKMode> prover::State<'a, F, MM> {
+    pub fn second_round(&mut self) -> Result<prover::SecondOracles<F>> {
         let round_time = start_timer!(|| "AHP::Prover::SecondRound");
 
         let zk_bound = Self::zk_bound();
 
-        let max_constraint_domain = state.max_constraint_domain;
+        let max_constraint_domain = self.max_constraint_domain;
 
-        let verifier::FirstMessage { batch_combiners, .. } = verifier_message;
-
-        let h_0 = Self::calculate_rowcheck_witness(&mut state, batch_combiners)?;
+        let h_0 = self.calc_rowcheck_witness()?;
 
         assert!(h_0.degree() <= 2 * max_constraint_domain.size() + 2 * zk_bound.unwrap_or(0) - 2);
 
         let oracles = prover::SecondOracles { h_0: LabeledPolynomial::new("h_0", h_0, None, None) };
-        assert!(oracles.matches_info(&Self::second_round_polynomial_info()));
+        assert!(oracles.matches_info(&AHPForR1CS::<F, MM>::second_round_polynomial_info()));
 
         end_timer!(round_time);
 
-        Ok((oracles, state))
+        Ok(oracles)
     }
 
-    fn calculate_rowcheck_witness(
-        state: &mut prover::State<F, MM>,
-        batch_combiners: &BTreeMap<CircuitId, verifier::BatchCombiners<F>>,
-    ) -> Result<DensePolynomial<F>> {
-        let mut job_pool = ExecutionPool::with_capacity(state.circuit_specific_states.len());
-        let max_constraint_domain = state.max_constraint_domain;
+    fn calc_rowcheck_witness(&mut self) -> Result<DensePolynomial<F>> {
```

Comment:
I guess we don't want to change these method *names*, right?

Reply:
f7c487afb